docs/spec/spec.md

```diff
@@ -0,0 +1,158 @@
+
+The conforming implementation of the specification is released and included in the distribution. Any deviation from the specification is considered a bug.
+
+## Contents
+
+1. [Overview](#1-overview)
+2. [Client](#2-client)
+    * 2.1 [Initializing the Client](#21-initializing-the-client)
+3. [`Profile` Resource](#3-profile-resource)
+4. [`Message` Resource](#4-message-resource)
+5. [`Draft` Resource](#5-draft-resource)
+6. [`MailThread` Resource](#6-mailthread-resource)
+7. [`Label` Resource](#7-label-resource)
+8. [`History` Resource](#8-history-resource)
+ 
+## 1. Overview
+
+Ballerina language offers first-class support for writing network-oriented programs. The Gmail package leverages these language features to create a programming model for consuming the Gmail REST API.
+
+The Gmail package offers intuitive resource methods to interact with the [Gmail API v1](https://gmail.googleapis.com/$discovery/rest?version=v1).
+
+## 2. Client
+
+This section outlines the client of the Ballerina Gmail package. To utilize the Ballerina Gmail package, a user must first import it.
+
+###### Example: Importing the Gmail Package
+
+```ballerina
+import ballerinax/googleapis.gmail;
+```
+
+The `gmail:Client` allows you to connect to the Gmail RESTful API. The client currently supports processing of `Profile`, `Message`, `Draft`, `Thread`, and `Label` resources. The client employs HTTP as the underlying protocol for communication with the API.
+
+#### 2.1 Initializing the Client
+
+The `gmail:Client` initialization method requires valid authentication credentials.
+
+```ballerina
+gmail:Client gmailClient = check new gmail:Client (
+        config = {
+            auth: {
+                refreshToken: refreshToken,
+                clientId: clientId,
+                clientSecret: clientSecret
+            }
+        }
+    );
+```
+
+The `gmail:Client` uses `http:Client` as its underlying implementation. You can configure this `http:Client` by providing the `gmail:ConnectionConfig` as a parameter during the `gmail:Client` initialization.
+
+## 3. `Profile` Resource
+
+You can retrieve the details of the authenticated user using the `/users/[userId]/profile()` resource method. The `Profile` record includes the following fields:
+
+```ballerina
+public type Profile record {
+    string emailAddress?;
+    string historyId?;
+    int:Signed32 messagesTotal?;
+    int:Signed32 threadsTotal?;
+};
+```
+
+###### Example: Retrieving Gmail profile 
```

Comment:
```suggestion
###### Example: Retrieving the Gmail profile 
```

Reply:
It is recommended to not go up to level 6 headings if possible.

Reply:
This is to just get the links to example. If this is not a heading we cannot share.